plugins/kubernetes/app/models/kubernetes/resource.rb

```diff
@@ -152,16 +157,20 @@ class ConfigMap < Base
     end
 
     class Service < Base
-      # ideally we should update, but that is not supported
-      # and delete+create would mean interrupting service
-      # TODO: warn users when a change needs to be made but could not be done
-      def deploy
-        return if running?
-        create
-      end
+      private
 
-      def revert(previous)
-        delete unless previous
+      def template_for_update
+        new = @template.deep_dup
+        [
+          "metadata.resourceVersion",
+          "spec.clusterIP",
+          *ENV["KUBERNETES_SERVICE_PERSISTENT_FIELDS"].to_s.split(",")
+        ].each do |keep|
+          path = keep.split(".").map!(&:to_sym)
+          old_value = resource.dig(*path)
+          new.dig_set path, old_value unless old_value.nil? # boolean fields are kept, but nothing is nil in kubernetes
+        end
+        new
```

Comment:
Would be a good idea to add a comment in here explaining why this is necessary for Service objects, and why the behavior is different than something like a Deployment.